cmake/OpenCVFindLAPACK.cmake

```diff
@@ -112,11 +112,15 @@ macro(ocv_lapack_check)
       execute_process(COMMAND sw_vers -productVersion
                       OUTPUT_VARIABLE MACOS_VERSION
                       OUTPUT_STRIP_TRAILING_WHITESPACE)
-      # Enable Accelerate New LAPACK if macOS >= 13.3
-      if (MACOS_VERSION VERSION_GREATER "13.3" OR MACOS_VERSION VERSION_EQUAL "13.3")
-        set(LAPACK_TRY_COMPILE_DEF "-DACCELERATE_NEW_LAPACK")
-        add_compile_definitions(ACCELERATE_NEW_LAPACK)
-        add_compile_definitions(ACCELERATE_LAPACK_ILP64)
+      # Compatibility for deployment on old macOS
+      if(${CMAKE_OSX_DEPLOYMENT_TARGET} VERSION_GREATER "13.3" OR ${CMAKE_OSX_DEPLOYMENT_TARGET} VERSION_EQUAL "13.3")
+        # Enable Accelerate New LAPACK if macOS >= 13.3
+        if(MACOS_VERSION VERSION_GREATER "13.3" OR MACOS_VERSION VERSION_EQUAL "13.3")
```

Comment:
Looks like the check is redundant.  In case if `CMAKE_OSX_DEPLOYMENT_TARGET` match requirements, then all headers and libs are already available. Host system version is not important.

Reply:
Agreed. It's almost always wrong to inspect anything about the host system, it's bad for cross-compilation generally, on all platforms.

You should not call `sw_vers` at all.

Reply:
Thank you for pointing out the problem.